packages/nimble-components/src/table-column/number-text/tests/table-column-number-text.spec.ts

```diff
@@ -0,0 +1,480 @@
+                value: Number.POSITIVE_INFINITY,
+                expectedRenderedString: '∞'
+            },
+            {
+                name: 'NaN renders as NaN',
+                value: Number.NaN,
+                expectedRenderedString: 'NaN'
+            },
+            {
+                name: '-0 renders with negative sign',
+                value: -0,
+                expectedRenderedString: '-0'
+            },
+            {
+                name: '+0 renders without positive sign',
+                value: 0,
+                expectedRenderedString: '0'
+            },
+            {
+                name: 'without exponential notation limits to 4 digits with rounding decimals up',
+                value: 1.23456789,
+                expectedRenderedString: '1.23457'
+            },
+            {
+                name: 'without exponential notation limits to 4 digits with rounding decimals down',
+                value: 10.001122,
+                expectedRenderedString: '10.0011'
+            },
+            {
+                name: 'does not use exponential notation for 999,999',
+                value: 999999,
+                expectedRenderedString: '999,999'
+            },
+            {
+                name: 'uses exponential notation for 1,000,000',
+                value: 1000000,
+                expectedRenderedString: '1E6'
+            },
+            {
+                name: 'does not use exponential notation for 0.000001',
+                value: 0.000001,
+                expectedRenderedString: '0.000001'
+            },
+            {
+                name: 'uses exponential notation for 0.000000999',
+                value: 0.000000999,
+                expectedRenderedString: '9.99E-7'
+            },
+            {
+                name: 'does not show decimals for an integer value',
+                value: 16,
+                expectedRenderedString: '16'
+            },
+            {
+                name: 'does not add extra decimals',
+                value: -98.75,
+                expectedRenderedString: '-98.75'
+            },
+            {
+                name: 'converts numbers with large magnitudes to exponential notation',
+                value: -123456789.123456789,
+                expectedRenderedString: '-1.23457E8'
+            },
+            {
+                name: 'converts numbers with small magnitudes to exponential notation',
+                value: 0.000000123456789,
+                expectedRenderedString: '1.23457E-7'
+            },
+            {
+                name: 'MAX_SAFE_INTEGER + 9999 renders as expected',
+                value: Number.MAX_SAFE_INTEGER + 9999,
+                expectedRenderedString: '9.0072E15'
+            },
+            {
+                name: 'MIN_SAFE_INTEGER - 9999 renders as expected',
+                value: Number.MIN_SAFE_INTEGER - 9999,
+                expectedRenderedString: '-9.0072E15'
+            }
```

Comment:
instead of "as expected" maybe "as an exponential with most significant digits of MAX_SAFE_INTEGER" or something